proof_test.go

```diff
@@ -0,0 +1,120 @@
+package merkletree
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+)
+
+func computeLeafHash(ap *AuthenticationPath) (leafHash []byte) {
+	leaf := ap.Leaf()
+	if leaf.Commitment() != nil {
+		// user leaf node
+		leafHash = crypto.Digest(
+			[]byte{LeafIdentifier},              // K_leaf
+			[]byte(ap.TreeNonce()),              // K_n
+			[]byte(ap.Index()),                  // i
+			[]byte(util.IntToBytes(ap.Level())), // l
+			[]byte(leaf.Commitment()),           // commit(key|| value)
+		)
+	} else {
+		// empty leaf node
+		leafHash = crypto.Digest(
+			[]byte{EmptyBranchIdentifier},       // K_empty
+			[]byte(ap.TreeNonce()),              // K_n
+			[]byte(ap.Index()),                  // i
+			[]byte(util.IntToBytes(ap.Level())), // l
+		)
+	}
+	return
+}
+
+func computeProofHash(ap *AuthenticationPath, depth int) (leftHash, rightHash []byte) {
```

Comment:
the flow of this function is mostly the same as `RecomputeHash()` method of `MerkleTree`. It returns the hash of left and right sub tree which then are compared to the value of `leftHash` and `rightHash` of the `root` node. (https://github.com/coniks-sys/libmerkleprefixtree-go/blob/1a48b4084a8821d21c25510c6192e97dd2083dd8/proof_test.go#L94)